deploy-queue/src/main.rs

```diff
@@ -0,0 +1,374 @@
+        } else if deployment.finish_timestamp.is_none() {
+            DeploymentState::Running
+        } else if deployment.finish_timestamp.is_some() && deployment.finish_timestamp.unwrap() < OffsetDateTime::now_utc() - time::Duration::minutes(deployment.buffer_time.into()) {
+            DeploymentState::FinishedInBuffer
+        } else {
+            DeploymentState::Finished
+        }
+    }
+}
+
+const BUSY_RETRY: StdDuration = StdDuration::from_secs(5);
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let log_env = env_logger::Env::default().filter_or("DEPLOY_QUEUE_LOG_LEVEL", "info");
+    env_logger::Builder::from_env(log_env).init();
+    let args = cli::Cli::parse();
+
+    // Create a single database connection for all operations
+    let db_client = create_db_connection().await?;
+
+    // Run new migrations after connecting to DB
+    run_migrations(&db_client).await?;
+
+    match args.mode {
+        Mode::Start {
+            region,
+            component,
+            environment,
+            version,
+            url,
+            note,
+            concurrency_key,
+        } => {
+
+            // Insert deployment record into database
+            let deployment_id = insert_deployment_record(
+                &db_client,
+                Deployment {
+                   region,
+                   component,
+                   environment: environment.to_string(),
+                   version,
+                   url,
+                   note,
+                   concurrency_key,
+                   ..Default::default()
+                }
+            ).await.context("Failed to enqueue new deployment")?;
+
+            loop {
+                // Check for blocking deployments in the same region
+                let blocking_deployments = check_blocking_deployments(&db_client, deployment_id).await?;
+                
+                if blocking_deployments.is_empty() {
+                    info!("No blocking deployments found. Deployment can be started.");
+
+                    start_deployment(&db_client, deployment_id).await
+                        .context("Failed to start deployment")?;
+                    info!("Successfully started deployment with ID: {}", deployment_id);
+                    break;
+                } else {
+                    // Print information about blocking deployments
+                    info!("Found {} blocking deployment(s) with smaller queue positions:", 
+                        blocking_deployments.len());
+                    for pending_deployment in blocking_deployments {
+                        let deployment_state: DeploymentState = (&pending_deployment).into();
+                        let deployment_note = pending_deployment.url.or(pending_deployment.note).unwrap_or_default();
+                        info!("  - Deployment ID: {}, Component: {}, State: {}, Note: {}", pending_deployment.id, pending_deployment.component, deployment_state, deployment_note);
```

Comment:
Looking at this code again I've realized a few things about it that bug me a bit:
- url being set should not hide the note, especially if we print it after `Note: `
- we might want to include the version here as well
- this line becomes fairly long, maybe we want to shorten this down to something like this: `{id} {state as verb} {component}(@{version})(: ({note}) ({url}))`
- we might want to move generating this summary line into a function for `Deployment`, so that we can reuse it for the `info` subcommand

Things I'd like to be done from this comment are version, url and note all being included if they exist, the rest is optional/follow-up

Reply:
I have just looked, I think I wasn't quite clear. I meant the parts in parenthesis to be omitted if the value is missing. Just pushed a small change for this.